### PR TITLE
[REF] discuss: rewrite tests on mail

### PR DESCRIPTION
**Pull request by zel-odoo**
rewrite some test files which are no longer suitable.
task-2893661

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/test_mail/tests/test_mail_message.py [hunk 1]
```diff
@@ -330,11 +332,6 @@ def test_mail_message_access_search(self):
         messages = self.env['mail.message'].with_user(self.user_portal).search([('subject', 'like', '_ZTest')])
         self.assertFalse(messages)
 
-        # Test: Portal: 2 messages (public group with a subtype)
-        self.group_pigs.write({'public': 'public'})
-        messages = self.env['mail.message'].with_user(self.user_portal).search([('subject', 'like', '_ZTest')])
-        self.assertEqual(messages, msg4 | msg5)
-
```
Comment (seb-odoo) [inline addons/test_mail/tests/test_mail_message.py:337]: Shouldn't we keep the assert with portal?

File: addons/test_mail/tests/test_mail_message.py [hunk 2]
```diff
@@ -363,46 +360,45 @@ def test_mail_message_access_read_notification(self):
...
+        self.message.write({'model': 'mail.channel', 'res_id': self.test_channel_everyone.id})
+        # Test: Employee reads the message, ok because linked to a doc they are allowed to read
         self.message.with_user(self.user_employee).read()
 
     # --------------------------------------------------
     # CREATE
     # --------------------------------------------------
 
-    @mute_logger('odoo.addons.base.models.ir_model')
+    @mute_logger('odoo.addons.base.models')
```
Comment (seb-odoo) [inline addons/test_mail/tests/test_mail_message.py:379]: why the change?

**Review by seb-odoo — COMMENTED** [review]
Some preliminary remarks, but it's not very easy to review.

Maybe split in more commits/PR to ease the transition.

1. just rename what needs to be renamed
2. change setup (if necessary) and adapt related assert
3. add new assertions

File: addons/test_mail/tests/test_mail_message.py (hunk 2 above)
Comment (zel-odoo) [reply addons/test_mail/tests/test_mail_message.py:379]: Becasue it's actually `odoo.addons.base.models.ir_rule` and  `odoo.addons.base.models.ir_model` producing the loggers. -> The public user not allowed to enter group channel. I will make it more clear with exactly those two.

File: addons/mail/tests/test_mail_channel.py [hunk 3]
```diff
@@ -45,113 +45,199 @@ def setUpClass(cls):
         cls.test_group = Channel.create({
             'channel_type': 'group',
             'name': 'Group',
+            'public': 'private'})
+        # Chat
+        cls.test_chat = Channel.create({
```
Comment (seb-odoo) [inline addons/mail/tests/test_mail_channel.py:50]: name suggestion: `chat_user1(_user2)` (where user(s) are specified, we shouldn't create an empty chat)

possibly use `channel_get` to create it

File: addons/mail/tests/test_mail_channel.py [hunk 4]
```diff
@@ -27,88 +27,88 @@ def setUpClass(cls):
...
+        # Channel for certain group
+        cls.test_channel_group = Channel.create({
+            'name': 'Channel for Group',
             'public': 'groups',
             'group_public_id': cls.env.ref('base.group_user').id})
-        # Jobs: public group
-        cls.group_public = Channel.create({
-            'name': 'Jobs',
+        # Public Channel
+        cls.test_channel_public = Channel.create({
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/tests/test_mail_channel.py:38]: In the setup commit: possibly use `channel_create` to create it

File: addons/mail/tests/test_mail_channel.py [hunk 5]
```diff
@@ -27,88 +27,88 @@ def setUpClass(cls):
         cls.user_public = mail_new_test_user(cls.env, login='user_public', groups='base.group_public', name='Bert Tartignole')
         cls.user_portal = mail_new_test_user(cls.env, login='user_portal', groups='base.group_portal', name='Chell Gladys')
 
-        # Pigs: base group for tests
-        cls.group_groups = Channel.create({
-            'name': 'Pigs',
+        # Channel for certain group
+        cls.test_channel_group = Channel.create({
```
Comment (seb-odoo) [inline addons/mail/tests/test_mail_channel.py:31]: In the setup commit: possibly use `channel_create` to create it

File: addons/mail/tests/test_mail_channel.py [hunk 6]
```diff
@@ -50,97 +55,166 @@ def test_access_public(self):
...
+        with self.assertRaises(AccessError):
+            self.env['mail.channel'].browse(self.chat_user_employee.id).unlink()
 
-    @mute_logger('odoo.addons.base.models.ir_rule', 'odoo.models')
+    @mute_logger('odoo.addons.base.models.ir_rule', 'odoo.addons.base.models.ir_model', 'odoo.models')
     @users('user_portal')
     def test_access_portal(self):
+        # Read public channel -> ok
+        self.env['mail.channel'].browse(self.public_channel.id).read()
+        # Read channel for gruop/group -> ko, ko, no access rights
```
Comment (seb-odoo) [inline addons/mail/tests/test_mail_channel.py:160]: typo: group

**Review by seb-odoo — COMMENTED** [review]
It looks ok, as this PR is only introducing asserts based on current access rules.
Some of the rules I find strange, but I suppose this is the right opportunity to see what improves in this regard with upcoming PRs.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by zel-odoo** [comment]
> @robodoo r+ rebase-ff

i will rebuild it
